community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/participant/SchemaIndexMigrator.java

```diff
@@ -72,8 +70,8 @@ public void moveMigratedFiles( File migrationDir, File storeDir, String versionT
         if ( deleteObsoleteIndexes )
         {
             deleteIndexes( schemaIndexDirectory );
-            deleteIndexes( labelIndexDirectory );
         }
+        deleteIndexes( LuceneLabelScanDirectory.getLuceneStoreDirectory( storeDir ) );
```

Comment:
I think deleting Lucene label scan store during migration is absolutely the correct thing to do, since we would need to delete it anyway on next startup.

The question is if we need to rebuild native label scan store as part of migration or not and I think that would be the best thing to do. Although I think it is outside the scope of this PR.